xapian-core/weight/tfidfweight.cc

```diff
@@ -63,16 +59,21 @@ TfIdfWeight::TfIdfWeight(const std::string &normals, double slope, double delta)
 	!strchr("ntpfsP", normalizations[1]) ||
 	!strchr("nP", normalizations[2]))
 	throw Xapian::InvalidArgumentError("Normalization string is invalid");
+    if (param_slope <= 0)
+	throw Xapian::InvalidArgumentError("Parameter slope is invalid.");
+    if (param_delta <= 0)
+	throw Xapian::InvalidArgumentError("Parameter delta is invalid.");
     if (normalizations[1] != 'n') {
 	need_stat(TERMFREQ);
 	need_stat(COLLECTION_SIZE);
     }
     need_stat(WDF);
     need_stat(WDF_MAX);
-    if (param_slope != 0 || param_delta != 0 || normalizations[2] == 'P') {
+    if ((param_slope > 0 || param_delta > 0) && normalizations[2] == 'P') {
```

Comment:
Okay, got it; dropping `param_slope > 0` and `param_delta > 0` checks and leaving just the final check here.